### DISCUSSION TITLE
[STRING] split ignores \n if lead by \r

### DISCUSSION THREAD

**Discussion by n3o77** [symfony/symfony#49515] [Q&A]
### Symfony version(s) affected

6.2.6

### Description

Not sure if really a bug or intended, but caught me off guard. I wanted to use UnicodeString instead of an explode with some other transformations and in the end split the string into pieces. But UnicodeString doesn't split the `\n` if it is lead by `\r`. 

Original code:
```php
$str = "A\r\nB\r\nC\nD";
array_map('trim', explode("\n", $str)); // ["A", "B", "C", "D"]
```
to:
```php
$str = "A\r\nB\r\nC\nD";
array_map('trim', u($str)->split("\n")); // ["A\r\nB\r\nC", "D"]
```

### How to reproduce

```php
$str = "A\r\nB\r\nC\nD";
u($str)->split("\n");
```

**Comment by nicolas-grekas** [comment]
Split expects a regexp, not a regular string.
You should use `u($str)->split('/\n/');` instead.

**Reply by n3o77** [reply]
Ok, makes sense. I didn't get that when looking at the docs nor the method description. But even with `u($str)->split('/\n/');` i get the same result.

**Reply by nicolas-grekas** [reply]
Hum, nvm, what I said is wrong, you're hitting a special behavior: `\r\n` is consider as a single unit (it's a grapheme) so it cannot be split.
You can work around this by using `$u->split("/\n/", null, 0)` (the `0` forces the regexp match, while the default is to match the string, unlike what I said in my first answer.)
